src/app/clusters/channel-server/channel-server.cpp

```diff
@@ -122,20 +122,30 @@ CHIP_ERROR ChannelAttrAccess::Read(const app::ConcreteReadAttributePath & aPath,
     EndpointId endpoint = aPath.mEndpointId;
     Delegate * delegate = GetDelegate(endpoint);
 
-    if (isDelegateNull(delegate, endpoint))
```

Comment:
questiom : there is no method to say isList ?   and then launch aEncoder.EncodeEmptyList();

Reply:
There is, sort of: you could query the metadata and then examine its type.

But fundamentally this code is just buggy: I would think that no delegate would mean you return an error!  For now I'm just preserving the existing behavior, so tests will keep passing, but that behavior is wrong....